src/irc/core/irc-cap.c

```diff
@@ -79,73 +79,148 @@ static void cap_emit_signal (IRC_SERVER_REC *server, char *cmd, char *args)
 	g_free(signal_name);
 }
 
+static gboolean parse_cap_name(char *name, char **key, char **val)
+{
+	const char *eq;
+
+	g_return_val_if_fail(name != NULL, FALSE);
+	g_return_val_if_fail(name[0] != '\0', FALSE);
+
+	eq = strchr(name, '=');
+	/* KEY only value */
+	if (eq == NULL) {
+		*key = g_strdup(name);
+		*val = NULL;
+		return TRUE;
+	/* Some values are in a KEY=VALUE form, parse them */
+	} else if (eq[1] != '\0') {
+		*key = g_strndup(name, (gsize)(eq - name));
+		*val = g_strdup(eq + 1);
+		return TRUE;
+	/* If the string ends after the '=' consider the value
+	 * as invalid */
```

Comment:
I dunno about this, a blank value could be meaningful. The spec doesn't consider this case but it doesn't say it should be rejected either.

Reply:
My reasoning was that you either provide a value or you don't, the specification isn't very clear with it's `name[=value]` description and I'd suggest to raise this issue with the irc3x folks.

Reply:
There's this: https://github.com/ircv3/ircv3-specifications/issues/334 - it's actually about message-tags, but people seem to agree that the solution should be the same for cap values too.

I like the idea of parsing `a=` as `key="a" value=""` (and `a` as `key="a" value=NULL`), it's fairly natural, so, just remove the special case imo

FWIW, dan- is working on a rewrite of the cap specs to make the whole thing less confusing, he's aware of this now.